### PR TITLE
Signup to barefoot using email&password and user verification

### PR DESCRIPTION
**Pull request by muhireheir**
creatingUser

> check your inbox to get verification link
>  'because your testing loccaly you have to change the url in your in box to this `http://localhost:port/api/v1/users/verify/token'
> -send a get request using postman or your browser
verify

Thank you !!

### CODE REVIEW

File: src/controllers/usersController.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      const hashedPassword = await bcrypt.hash(req.body.password, 10);
+
+      const newUser = {
+        email: req.body.email,
+        password: hashedPassword,
+      };
+      const createdUser = await userService.createuser(newUser);
+      return sendLink(res, createdUser);
+    } catch (error) {
+      util.setError(500, error.message, null);
```
Comment (nkalyesubula) [inline src/controllers/usersController.js]: setError takes in only two parameters

File: src/helpers/mailHelper.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import nodemailer from 'nodemailer';
+
+export const transporter = nodemailer.createTransport({
+  service: 'gmail',
+  auth: {
+    user: 'muhire416@gmail.com',
```
Comment (nkalyesubula) [inline src/helpers/mailHelper.js]: create a new gmail account that's specifically for barefoot-nomad-phoenix. And put the credentials in the .env

File: src/helpers/tokenVerifier.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import jwt from 'jsonwebtoken'
+export const decodeToken=async(token)=>{
+return  await jwt.verify(token,process.env.JWTKEY);
+}
```
Comment (nkalyesubula) [inline src/helpers/tokenVerifier.js]: This should be in the middleware folder

File: src/services/templates/joiErrorTemplate.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        example: 'xxx@yyy.zzz',
+      };
+      util.setError(400, Error);
+      return util.send(res);
+    }
+    if (
+      error.details[0].message
+        .replace('/', '')
+        .replace(/"/g, '')
+        .includes('fails to match the required')
```
Comment (nkalyesubula) [inline src/services/templates/joiErrorTemplate.js]: templates folder should only contain html code snippets. Put this code in the helpers folder

**Review by nkalyesubula — CHANGES_REQUESTED** [review]
Good work so far. In addition to the comments i've left. Please also address the following
1. Polish up your PR title
2. Super admin should be put in the seeds thereby having him created at the point of starting the app
3. Fix failing tests

**Review by Faustin250 — APPROVED** [review]
Good, done a great job at concern separations!

**Review by bizip — CHANGES_REQUESTED** [review]
I am trying to test this PR locally but I am blocked by the following error
Screen Shot 2020-10-21 at 12 48 58

**Comment by bizip** [comment]
> I am trying to test this PR locally but I am blocked by the following error
> Screen Shot 2020-10-21 at 12 48 58

**Review by bizip — APPROVED** [review]

**Review by nkalyesubula — APPROVED** [review]

**Pull request merged**